third_party/move/move-model/src/model.rs

```diff
@@ -1686,34 +1689,75 @@ impl GlobalEnv {
             };
 
             // While releasing any mutation, compute the used/called functions if needed.
-            let fun_data = &self.module_data[module_id.to_usize()]
-                .function_data
-                .get(&fun_id)
-                .unwrap();
-            let used_funs = if fun_data.used_funs.is_none() {
-                Some(self.get_used_funs_from_bytecode(&module, def_idx))
+            let (used_funs, called_funs) = if let Some(fun_data) = self.module_data
+                [module_id.to_usize()]
+            .function_data
+            .get(&fun_id)
+            {
+                let used_funs = if fun_data.used_funs.is_none() {
+                    Some(self.get_used_funs_from_bytecode(&module, def_idx))
+                } else {
+                    None
+                };
+                let called_funs = if fun_data.called_funs.is_none() {
+                    Some(self.get_called_funs_from_bytecode(&module, def_idx))
+                } else {
+                    None
+                };
+                (used_funs, called_funs)
             } else {
-                None
+                let used_funs = Some(self.get_used_funs_from_bytecode(&module, def_idx));
+                let called_funs = Some(self.get_called_funs_from_bytecode(&module, def_idx));
+                (used_funs, called_funs)
             };
-            let called_funs = if fun_data.called_funs.is_none() {
-                Some(self.get_called_funs_from_bytecode(&module, def_idx))
-            } else {
-                None
+            let (default_loc, fun_source_map_def_loc_opt) = {
```

Comment:
Could we add a top level comment like the above block (which is computing used/called funs) about what this next block of code does?